### ISSUE TITLE
[FEA] astype as a dataframe method

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#2195]
We should implement `astype` as a dataframe method, consistent with pandas.

When the entire dataframe cannot be cast, pandas will raise an error by default. Optionally, the error can be ignored and the resulting dataframe will have only cast the columns that can be successfully cast (leaving the other columns as is). See the third snippet for an example.

[This](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.astype.html) is the relevant pandas method. We could potentially implement this with `_apply_support_method` and some additional logic.

```python
import pandas as pd

df = pd.DataFrame({'a':range(3), 'b':range(3)})
df.astype('float')
	a	b
0	0.0	0.0
1	1.0	1.0
2	2.0	2.0
```

```python
import pandas as pd

df = pd.DataFrame({'a':range(3), 'b':['a','b','c']})
df.astype('object')
   a  b
0  0  a
1  1  b
2  2  c
```

```python
import pandas as pd

df = pd.DataFrame({'a':range(3), 'b':['a','b','c'], 'c':range(3,6)})
print(df.dtypes)
df.astype('float', errors='ignore').dtypes
a     int64
b    object
c     int64
dtype: object
a    float64
b     object
c    float64
dtype: object
```

**Comment by brandon-b-miller** [comment]
Can do 👍

**Comment by kkraus14** [comment]
Fixed by #3022
